src/service.rs

```diff
@@ -779,7 +786,7 @@ where
                             &mut self.service_context,
                             ServiceError::ListenError {
                                 error: Error::RepeatedConnection(context.inner.id),
-                                address,
+                                address: listen_addr.expect("listen address must be exist"),
```

Comment:
```suggestion
                                address: listen_addr.expect("listen address must exist"),
```

Reply:
Why we can ensure `listen_addr` is not None, I see several places that call `session_open` or `handshake` with None.

Reply:
There are two situations for a handshake:
1. self is listener, session type is inbound
2. self is dialer, session type is outboud

When self is listener, session is inbound, listen address must exist

Reply:
Is it possible to change `SessionType::Inbound` to ` SessionType::Inbound { listen_addr: MultiAddr }` ?

Reply:
If not, we can add an assert that when session type is inbound, `listen_addr` must be Some.

Reply:
It's a breaking change. 

Users need to know which listen port the session comes from?

Reply:
> If not, we can add an assert that when session type is inbound, `listen_addr` must be Some.

I think `expect` is the same as assert on here